src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+    );
+
+    let errorText;
+    if (enteredFutureTime) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot begin or end in the future.
+      </Typography>
+      );
+    } else if (timeOutIsBeforeTimeIn) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot end before it starts.
+      </Typography>
+      );
+    } else if (shiftTooLong) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot be longer than 20 hours.
+      </Typography>
+      );
+    } else if (shiftTooShort) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot be shorter than 15 minutes.
+      </Typography>
+      );
+    } else if (isOverlappingShift) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          The entered shift conflicts with a previous shift.
+      </Typography>
+      );
+    }
+    else {
+      errorText = <></>;
+    }
+
+    const onDatetimeSelectorClose = () => {
+      setIsOverlappingShift(false);
+      if (selectedDateIn !== null && selectedDateOut !== null) {
+        getActiveJobsForUser();
+        checkForFutureDate();
+      }
+    }
+
+    const handleShiftNotesChanged = event => {
+      setUserShiftNotes(event.target.value);
+    };
+
+    return (
+      <>
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+              <Card>
+                <CardContent
+                  style={{
+                    marginLeft: 8,
+                    marginTop: 8,
+                  }}
+                >
+                  <CardHeader title="Enter a shift" />
+                  <Grid
+                    container
+                    spacing={2}
+                    justify="space-around"
+                    alignItems="center"
+                    alignContent="center"
+                  >
+                    <Grid item xs={12} sm={6} md={3}>
+                      <DatePicker
+                        autoOk
+                        variant="inline"
+                        margin="normal"
+                        id="date-picker-in-dialog"
+                        label="Date In"
+                        format="MM/dd/yyyy"
+                        value={selectedDateIn}
+                        onChange={handleDateChange1}
+                        onClose={onDatetimeSelectorClose}
+                      />
+                    </Grid>
+                    <Grid item xs={12} sm={6} md={3}>
+                      <TimePicker
+                        variant="inline"
+                        margin="normal"
+                        id="time-picker-in"
+                        label="Time In"
+                        value={selectedDateIn}
+                        onChange={date => {
+                          let dateToChange = date;
```

Comment:
Can this just be `onChange={handleDateChange2}`, as was done on 433?

Reply:
It can't be `handleDateChange2` because that is specific to date out, while this one is specific to date in. This touches on the need to rename those two functions again. 

One thing I'm not sure about: why did we do the whole `let dateToChange = date;                           handleDateChange1(dateToChange);` instead of `handleDateChange1(date);`? Beats me. We will look into this.

Reply:
Oh, you noticed that! I was wondering the same. Did you have a reason?

Reply:
And just as I post this I see the change where you replaced it- sorry you had to hear me harping on it after you fixed it!